ZelBack/src/services/fluxNetworkHelper.js

```diff
@@ -254,12 +255,40 @@ async function checkAppAvailability(req, res) {
   });
   req.on('end', async () => {
     try {
+      const authorized = await verificationHelper.verifyPrivilege('adminandfluxteam', req);
+
+      const remoteIP = req.ip || req.connection.remoteAddress || req.socket.remoteAddress || req.headers['x-forwarded-for'];
+
+      const remoteIP4 = remoteIP.replace('::ffff:', '');
+
       const processedBody = serviceHelper.ensureObject(body);
 
-      const { ip, ports, appname } = processedBody;
+      const {
+        ip, ports, appname,
+      } = processedBody;
+      const apiport = processedBody.apiport || config.server.apiport;
+
+      if (ip !== remoteIP4 && authorized === false) {
+        throw new Error(`Request ip ${remoteIP4} of ${remoteIP} doesn't match the ip: ${ip} to check App port Availability.`);
+      }
+
+      if (authorized === false) { // can only be called from a valid fluxnode if not authorized call
+        const nodeList = await fluxCommunicationUtils.deterministicFluxList();
+        let ipPortOnNodeList = ip;
+        if (apiport !== '16127' && apiport !== 16127) {
```

Comment:
This is wrong, the api port is from the node calling, not from the node where this is being executed. Because it was added as a new property on the post, it cannot be used here until all nodes are updated. That is why I left this part commented.

Reply:
the apiport should come from:
const {
        ip, apiport, ports, appname,
      }
but only updated nodes will send that information.

Reply:
I think it is fine to leave it enabled - majority of the nodes upgrade very fast to the latest release making it a non issue for those nodes.
If apiport is not supplied we fall back to the default one basically searching just an IP in the deterministic list. Which usually is present if people are using upnp anyway. This makes it fine as well.
Ideally we should be submitting public key and signature if we really want to make this very strict but I think above ip check will serve the needs very well.

But I am actually seeing a different issue now. As unconfirmed nodes will basically get dossed immediately as they are not on the confirmed list. 
I think we have to add the signature to it actually to be 100% sure.